qip-0015.md

```diff
@@ -0,0 +1,59 @@
+Because the UTXO set can grow to be infinitely large, a well-funded attacker could launch a DDoS attack to grow the UTXO set to a size that is untenable for the average node to store on disk. In order to prevent this attack, this QIP proposes burning old, unused small-value UTXOs (below 1 Qi), such that the DDoS bloating attack would be expontentially more expensive (10 billion Qi vs 10 million Qi, for example). 
+
+## Specification
+
+### UTXO Trimming
+
+The following table displays small-value denominations and the timeframe after which each respective denomination should be trimmed (removed from the set). Note that denomination 0 is equivalent to 0.001 Qi or 1 "Qit" (the smallest unit of Qi).
+| Denomination      | Value                 | Timeframe                         
+|-------------------|-----------------------|----------------------------------
+| 0                 | 0.001 Qi (1 "Qit")    | 2 weeks                      
+| 1                 | 0.005 Qi (5 "Qits")   | 1 month                   
+| 2                 | 0.01 Qi (10 "Qits")   | 2 months 
+| 3                 | 0.05 Qi (50 "Qits")   | 4 months          
+| 4                 | 0.1 Qi (100 "Qits")   | 8 months         
+| 5                 | 0.25 Qi               | 10 months         
+| 6                 | 0.5 Qi                | 12 months 
+
+Every block will trim UTXOs created by a block at a depth that is determined to be approximately equal to each timeframe. That means that every block will trim UTXOs of a certain denomination from 6 different blocks. This can be compute intensive, so it's recommended that the implementation perform the trimming of each block in parallel, which requires a CPU with 7 threads at minimum.
+The trimming must happen synchronously as part of the block append process, as the result of the trimming process modifies the UTXO set commitment and must be followed by every node (but not necessarily in a specific order, which is a property unique to the MuHash commitment).
+
+### UTXO Key Organization
+Because we must trim blocks at depth, we must store the UTXOs created by each block on disk. The UTXO set is a flat database, not keyed by block hash, but rather keyed by transaction hash and index. One way to do this is to add the block number and denomination into the key as a prefix, and then iterate through all keys on disk with the block number and denomination as the prefix and trim the UTXOs under that prefix.
+The issue with this approach is that it requires the user or sender of a transaction to provide the block number and denomination into the input so the UTXO can be looked up for verification. Denomination is easy, as the user knows how much they will spend, but block number is slightly more difficult, as the block height that created a UTXO can change at the tip. A wallet might mistakenly mark that a UTXO was created in a certain block, only for that information to be invalidated 5 minutes later.
```

Comment:
> but block number is slightly more difficult

this is an understatement. Its practically impossible.

> 5 minutes later

or 0.1 seconds later